core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -156,13 +158,73 @@ public ACL getACL() {
      *      just needs to return {@link PlannedNode}s that each contain an object that implements {@link Future}.
      *      When the {@link Future} has completed its work, {@link Future#get} will be called to obtain the
      *      provisioned {@link Node} object.
+     * @deprecated Use {@link #provision(CloudState, int)} instead.
      */
-    public abstract Collection<PlannedNode> provision(Label label, int excessWorkload);
+    @Deprecated
+    public Collection<PlannedNode> provision(Label label, int excessWorkload){
+        try {
+            // Check if the new method is implemented
+            getClass().getDeclaredMethod("provision", CloudState.class, int.class);
```

Comment:
Found a problem with this aproach if there are several levels of subclassing. I wonder if there is already a utility covering this case in the Jenkins codebase...

Reply:
`Util.isOverridden`